pkg/kubelet/images/image_manager.go

```diff
@@ -155,6 +205,24 @@ func (m *imageManager) EnsureImageExists(pod *v1.Pod, container *v1.Container, p
 	}
 	m.logIt(ref, v1.EventTypeNormal, events.PulledImage, logPrefix, fmt.Sprintf("Successfully pulled image %q in %v", container.Image, time.Since(startTime)), klog.Info)
 	m.backOff.GC()
+
+	if imagePullResult.hash == "" {
+		// successful pull no auth hash returned, auth was not required so we should reset the hashmap for this
+		// imageref since auth is no longer required for the local image cache, allowing use of the ImageRef
+		// by other pods if it remains cached and pull policy is PullIfNotPresent
+		ensureSecretPulledImage[imageRef] = nil
+	}
+	// store/create hashMatch map entry for auth config hash key used to pull the image
+	// for this imageref (digest)
+	if imagePullResult.hash != "" {
```

Comment:
nit: could be moved up to else branch of previous block.

Reply:
will do